src/Types.hs

```diff
@@ -462,8 +463,9 @@ data Mode =
     | PostListOverlay PostListContents
     deriving (Eq)
 
--- | We're either connected or we're not.
-data ConnectionStatus = Connected | Disconnected
+-- | We're either connected or we're not. If connected, keep the latest
+-- | websocket action sequence number and the websocket itself.
+data ConnectionStatus = Connected Int64 MMWebSocket | Disconnected
```

Comment:
I'd like to avoid keeping the websocket in the state and then having to update it. Instead I'd rather have a `TChan` in the state that can be used to communicate with whatever is managing the websocket. Would you be willing to refactor this so that's the case, and so that `Connected` has no arguments?

Reply:
There needs to be a new thread running to watch the `TChan` in that case, right? Is that okay? If so, I can refactor this.

Reply:
The websocket sequence number is supposed to be a monotonically increasing number. From the mattermost documentation [here](https://api.mattermost.com/#tag/WebSocket), "The seq or sequence number is set by the client and should be incremented with every use. It is used to distinguish responses to requests that come down the WebSocket."

If this is refactored to use a separate thread to watch the `TChan` as you mentioned, we can more this sequence number to that thread instead of keeping it in the `ChatState`.

Reply:
Yeah, I think that would be a nice change. The code in `Connection.hs` is already pretty close to that. I imagine that `RefreshWebsocketEvent` might go away, making `Events.hs` no longer repsonsible for calling `connectWebsockets` since that'd be done by a websocket manager thread (that doesn't exist).

Reply:
> If this is refactored to use a separate thread to watch the TChan as you mentioned, we can more this sequence number to that thread instead of keeping it in the ChatState.

Yeah, that's what I had in mind. It hides that management detail from the rest of the program.

Reply:
Sure. I'll work on this and the other comments and update the PR.

Reply:
Thank you!